src/main/java/fr/insee/keycloak/provider/FranceConnectIdentityProvider.java

```diff
@@ -103,7 +104,23 @@ protected boolean verify(JWSInput jws) {
     @Override
     public BrokeredIdentityContext getFederatedIdentity(String response) {
         try {
-            return super.getFederatedIdentity(response);
+            BrokeredIdentityContext federatedIdentity = super.getFederatedIdentity(response);
+            // Code to add verification for eidas level
+            // See: https://github.com/InseeFr/Keycloak-FranceConnect/issues/29
+            JsonWebToken idToken = (JsonWebToken) federatedIdentity.getContextData().get("VALIDATED_ID_TOKEN");
+            if (idToken == null) {
+                throw new IdentityBrokerException("The id_token cannot be retrieved");
+            }
+            FranceConnectIdentityProviderConfig.EidasLevel acrReturned = FranceConnectIdentityProviderConfig.EidasLevel.getOrDefault((String) idToken.getOtherClaims().get("acr"), FranceConnectIdentityProviderConfig.EidasLevel.EIDAS1);
```

Comment:
Small question about names: in our code, we talk about eidas level, in id token this is 'acr'.
Which name should we use? 'eidasLevel' is a little more explicit

getOrDefault is not the most suitable here I think, if franceConnect sends us 'acr = test', we have EIDAS1 here :/
Maybe we can create a get method who returns null if not found.

Reply:
Depends on the context:
- OIDC standard defines "acr" standing for "Authentication Context Class Reference" which is a string specifying an Authentication Context Class Reference value that identifies the Authentication Context Class that the authentication performed satisfied
- It is usually a number that corresponds to the level of assurance (LoA)
- FC decides to put EIDAS level (eidas1, eidas2 and eidas3). It is its own implementation choice
Since we are in FranceConnect provider, it is not shoking to call it eidasLevel. I can change it to eidasLevelReturned and eidasLevelRequest in order to have the same naming everywhere

I put `getOrDefault((String) idToken.getOtherClaims().get("acr"), FranceConnectIdentityProviderConfig.EidasLevel.EIDAS1)` since in the /authorize request we assume that if `acr_values` does not match the enum, the default one is eidas1: see `getEidasLevel()` method. I found it coherent to make the same assumption for the response (/token). Maybe we shouldn't!

Reply:
Thanks for your explanation.
`acr_values` comes from html template file `realm-identity-provider-franceconnect-particulier.html` and cannot be wrong but if angular fails at `ng-init` it can be null, so I put a little check to return a default behaviour.
I also think that `acr_values` can be change by `eidas_values`, I will create a PR after yours.

Reply:
The key `acr_values` in the /authorize request cannot be changed to `eidas_value` because this naming is defined by OIDC standard: see https://openid.net/specs/openid-connect-core-1_0.html, section 3.1.2.1.
To avoid ambiguity, the **keys** `acr_values` and `acr`are defined by the OIDC standard and cannot be changed. The **values** of these keys `eidas1`, `eidas2`, and `eidas3` are defined by France Connect (its own implementation choice)

Reply:
That's true but I see this attribute name in a different way. I think we can use eidas_values in our business logic and map to acr_values field only in queryParam for the request. Maybe I'm wrong, I will check if it's possible.